internal/testutil/acc/atlas.go

```diff
@@ -0,0 +1,112 @@
+package acc
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+// TestMainExecution must be called from TestMain in the test package if ProjectIDExecution is going to be used.
+func TestMainExecution(m *testing.M) {
+	atlasInfo.init = true
+	atlasInfo.resourceName = resourceName()
+
+	exitCode := m.Run()
+
+	if atlasInfo.needsDeletion {
+		fmt.Printf("Deleting execution project: %s, resource: %s\n", atlasInfo.projectName, atlasInfo.resourceName)
+		deleteProject(atlasInfo.projectID)
+	}
+
+	os.Exit(exitCode)
+}
+
+// ProjectIDExecution returns a project id created for the execution of the resource tests.
+func ProjectIDExecution(tb testing.TB) string {
+	tb.Helper()
+	SkipInUnitTest(tb)
+	require.True(tb, atlasInfo.init, "TestMainExecution must called to be able to use ProjectIDExecution")
+
+	atlasInfo.mu.Lock()
+	defer atlasInfo.mu.Unlock()
+
+	// lazy creation so it's only done if really needed
+	if atlasInfo.projectName == "" {
+		var globalName, globalID string
+		if atlasInfo.resourceName != "" {
+			globalName = (prefixProjectKeep + "-" + atlasInfo.resourceName)[:projectNameMaxLen]
+			globalID = projectID(globalName)
+		}
+
+		if globalID == "" {
+			atlasInfo.projectName = RandomProjectName()
+			tb.Logf("Creating execution project: %s, resource: %s, global project (not found): %s\n", atlasInfo.projectName, atlasInfo.resourceName, globalName)
+			atlasInfo.projectID = createProject(tb, atlasInfo.projectName)
+			atlasInfo.needsDeletion = true
+		} else {
+			atlasInfo.projectName = globalName
+			tb.Logf("Reusing global project: %s, resource: %s\n", atlasInfo.projectName, atlasInfo.resourceName)
+			atlasInfo.projectID = globalID
```

Comment:
Is there a possibility of having a separate function for providing the global project? I believe this would help simplify this function and make its usage more clear. In that case I understand that `needsDeletion` property could be removed as well.

Reply:
i had in mind that projects just ask for a project and a new one or a resource-global one is given but they don't need  to know.  and there is a nice fallback mechanism so you can create global-resource clusters for the most demanding resources and maybe for resources that just create a couple of projects you don't want to create a global-resource one as the benefit is not so big and you need that project always created.
 
do you think it's better to remove this abstraction and projects decide if they want an execution project or a global-project?

Reply:
in any way there will be a way to ask for specific projects when they need some extra config like triggers, but i don't think it's so useful to allow this for regular projects